Update the OpenShift compatibility with 7.9 (#3751)
elastic/cloud-on-k8s: docs/advanced-topics/openshift.asciidoc

This page shows how to run ECK on OpenShift.

* <<{p}-openshift-before-you-begin>>
* <<{p}-openshift-deploy-the-operator>>
* <<{p}-openshift-deploy-elasticsearch>>
* <<{p}-openshift-deploy-kibana>>

> * <<{p}-openshift-apm>>

Reviewer: ```suggestion
* <<{p}-openshift-anyuid-workaround>>
```

* <<{p}-openshift-beats>>

> WARNING: For versions older than 7.9, APM Server and Enterprise Search are not compatible with `restricted` SCC. Use this <<{p}-openshift-apm,workaround>> to overcome this limitation.

Reviewer: I think we should let the user know that Enterprise Search 7.8 was compatible with Openshift. This is a specific bug in the 7.9 release. Otherwise they might be surprised to have something working with 7.8 and not with 7.9 ?

```suggestion
WARNING: Some Docker images are known to be incompatible with the `restricted` SCC. This is the case for the *APM Server before 7.9* and for the *7.9 release of Enterprise Search*. A <<{p}-openshift-anyuid-workaround,workaround>> is to run those images with the `anyuid` SCC.
```

> With version 7.9, the APM Server is compatible with `restricted` SCC, while for Enterprise Search the incompatibility remains and can be addressed with the same workaround.

Reviewer: ```suggestion
```

* <<{p}-openshift-apm,Deploy an APM Server instance with a route>>
* <<{p}-openshift-beats,Grant privileged permissions to Beats>>

> WARNING: With Elastic Stack 7.9, note that APM Server is now compatible with OpenShift, see <<{p}-openshift-apm>>. Enterprise Search 7.9 is not yet compatible with Openshift.

Reviewer: > Enterprise Search 7.9 is not yet compatible with Openshift

Enterprise Search before 7.9 was compatible with the `restricted` SCC . We have an incompatibility starting with 7.9.x  On a second thought I'm wondering if the same workaround used for APM < 7.9 can be used for Enterprise Search = 7.9.x.

I need to do the test, will update my review with my findings.

Reviewer: I just did the test and I can confirm that if a user really want to deploy Enterprise Search 7.9.x on Openshift then the same workaround as the one described for the APM Server < 7.9 can be applied.

The only thing I'm not sure about is how to reflect those 2 two particular cases and the workaround in the doc. I guess that if we want to broaden the scope of the workaround we have to:
* Update the title `Deploy an APM Server instance with a route` to something more generic like `Deploy Docker images with the anyuid SCC` 
* Highlight that the workaround only aims to circumvent incompatibilities with APM < 7.9 and Enterprise Search = 7.9.x

Author: OK, thank you @barkbay!

Author: @barkbay I reworked the Warning and turned the APM title to something more generic (as you suggested). Let me know if this looks any better.

----

> [id="{p}-openshift-apm"]

Reviewer: ```suggestion
[id="{p}-openshift-anyuid-workaround"]
```

[id="{p}-openshift-apm"]
== Deploy an APM Server instance with a route

> With Elastic Stack 7.9, it is possible to run the APM Server with the `restricted` SCC. For versions older than 7.9, you can use this workaround to allow the Pod to run with the default `uid 1000` by assigning it to the `anyuid` SCC:

Reviewer: ```suggestion
Starting with version 7.9, it is possible to run the APM Server with the `restricted` SCC. For versions older than 7.9, you can use this workaround to allow the Pod to run with the default `uid 1000` by assigning it to the `anyuid` SCC:
```
